include/ibamr/CIBStochasticMethod.h

```
@@ -0,0 +1,176 @@
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#ifndef included_IBAMR_CIBStochasticMethod
+#define included_IBAMR_CIBStochasticMethod
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+
+#include "RobinBcCoefStrategy.h"
+#include "ibamr/CIBMethod.h"
+#include "ibamr/CIBStrategy.h"
+#include "ibamr/IBMethod.h"
+#include "ibtk/LData.h"
+#include "ibtk/LDataManager.h"
+
+namespace mu
+{
+class Parser;
+} // namespace mu
+
+namespace IBTK
+{
+class HierarchyMathsOps;
+} // namespace IBTK
+namespace IBAMR
+{
+class CIBStandardInitializer;
+class CIBStaggeredStokesOperator;
+} // namespace IBAMR
+
+/////////////////////////////// CLASS DEFINITION /////////////////////////////
+
+namespace IBAMR
+{
+/*!
+ * \brief Class CIBFEMethod is a concrete CIBStrategy and IBMethod
+ * class which implements the motion of rigid bodies using the constraint
+ * formulation. The immersed structure is discretized using standard IB
+ * markers.
+ */
+
+class CIBStochasticMethod : public IBAMR::CIBMethod
+{
+public:
+    /*!
+     * \brief Constructor of the class.
+     */
+    CIBStochasticMethod(const std::string& object_name,
+                        SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> input_db,
+                        const int no_structures = 1,
+                        bool register_for_restart = true);
```

parameters need descriptions: what are the number of structures?

---

no_structures is the number of rigid bodies being simulated. I made it default to 1 but this is probably bad as it may hide a good/helpful error is this value is not properly being set in main.

---

@brennansprinkle / @amneetb, are you able to get the structure indexing information from `LDataManager`, or is this a different notion of "structures" than what `LDataManager` provides?

---

Addressing this also would affect `CIBMethod`. Let's do this in a separate PR. I've recorded the issue here: #539.